Feature/redux

gss78 opened the pull request.

Commit:
Install @reduxjs/toolkit package

Commit:
Install react-redux package

Commit:
Install redux package

Commit:
Connect redux to the project

greybutton commented on app/javascript/slices/TasksSlice.js:
```diff
@@ -0,0 +1,59 @@
+      dispatch(loadColumnSuccess({ ...data, columnId: state, action }));
+    });
+  };
+
```
лучше добавить отдельную функцию loadColumnMore которая будет похожа на loadColumn, но внутри будет объединение загруженного и текущего стейта. Сейчас replaceItems, mergeItems больше вопросов вызывают.  loadColumnMore даст семантики и контекста, да будет дублирование, но здесь лучше дублирование, чем плохая абстракция и параметры-флаги

gss78 replied:
посмотри plz 
https://github.com/gss78/TaskManager/pull/17/commits/dae6883f3dfbc855048ec7d26d82f48f7c9c0b2e

greybutton commented on app/javascript/App.js:
```diff
@@ -0,0 +1,13 @@
+import TaskBoard from 'containers/TaskBoard';
+
+const App = (user) => (
+  <Provider store={store}>
+    <TaskBoard user={user} />
+  </Provider>
```
mui theme не стал подключать?

gss78 replied:
да, не понял что за импорт, те откуда это взять

import MUITheme from 'MUITheme/MUITheme';

или имееться ввиду 

import { ThemeProvider } from "@material-ui/styles";

потом определение theme 
(или createMuiTheme что такое)

и 

```
<ThemeProvider `theme={theme}\>`
...
</ThemeProvider>

```
добавлю, просто хотелось бы уточнить.

greybutton replied:
самому создать файл `app/javascript/MUITheme/MUITheme.js` и в нем уже использовать ThemeProvider, createMuiTheme. В верном направлении код написал <ThemeProvider `theme={theme}\>`

gss78 replied:
посмотри plz
https://github.com/gss78/TaskManager/pull/17/commits/fd6a07a98941db0336b6a5d129b5183c3dacb934

Commit:
Refactor loadColumnMore function

Commit:
Convert datapicker date to string on save in order to fix the warning..

Warning: Failed prop type: Invalid prop `task.expiredAt` of type `date` supplied to `Form`, expected `string`.

greybutton commented on app/javascript/slices/TasksSlice.js:
```diff
@@ -0,0 +1,73 @@
+      page,
+      perPage,
+    }).then(({ data }) => {
+      const cards = { items: [...data.items], meta: { ...data.meta } };
+      const column = board.columns.find(propEq('id', state));
+
+      cards.items = [...column.cards, ...data.items];
```
`const cards = { items: [...data.items], meta: { ...data.meta } };`
выглядит как просто копирование, не очень понятно зачем и можно без мутации `cards.items = `

```
data = { items: [...column.cards, ...data.items], meta: data.meta, columnId: state }
loadColumnSuccess(data)
```

вообще лучше добавить в reducers `loadColumnMoreSuccess` и там это все сделать, тогда колонка один раз будет искаться и не нужен будет `const board = useSelector((state) => state.TasksSlice.board);`

то, что loadColumnMoreSuccess будет почти повторять loadColumnSuccess это ок

gss78 replied:
сложно не согласиться,
https://github.com/gss78/TaskManager/pull/17/commits/9d82f7ed4c29281b62018bf280b2dafa910762c9

Commit:
Refactor loadColumnMore function 2

Commit:
Add MUITheme component
